uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/leave-request-api.js

```diff
@@ -18,33 +18,32 @@ define([
        *   `page` for the current page, `size` for number of items per page
        * @param {string} sort - The field and direction to order by
        * @param  {Object} params
-       * @return {Promise}
+       * @return {Promise} Resolved with {Object} All leave requests
        */
       all: function (filters, pagination, sort, params) {
         $log.debug('LeaveRequestAPI.all');
 
-        return this.getAll('LeaveRequest', filters, pagination, sort, params, 'getFull')
-          .then(function (data) {
-            return data;
-          });
+        return this.getAll('LeaveRequest', filters, pagination, sort, params, 'getFull');
       },
 
       /**
        * This method returns all the total change in balance that is caused by the
        * leave requests of a given absence type, or of all the absence types of a given contact and period.
        *
-       * @param {string} contactId Mandatory The ID of the Contact to get the balance change for
-       * @param {string} periodId Mandatory The ID of the Absence Period to get the balance change for
-       * @param statuses {array} Optional (Default: null) An array of OptionValue values which the list will be filtered by
-       * @param isPublicHoliday {boolean} Optional (Default: false) Based on the value of this param,
+       * @param {string} contactId The ID of the Contact to get the balance change for
+       * @param {string} periodId The ID of the Absence Period to get the balance change for
+       * @param {array} [statuses = null] An array of OptionValue values which the list will be filtered by
+       * @param {boolean} [isPublicHoliday=false] Based on the value of this param,
        * the calculation will include only the leave requests that aren't/are public holidays
-       * @return {Promise}
+       * @return {Promise} Resolved with {Object} Balance Change data or Error data
        */
       balanceChangeByAbsenceType: function (contactId, periodId, statuses, isPublicHoliday) {
         $log.debug('LeaveRequestAPI.balanceChangeByAbsenceType');
-
-        if(!contactId || !periodId) {
-          throw new Error("contact_id and period_id should have truthy value");
+        var deferred = $q.defer();
+        if (!contactId || !periodId) {
```

Comment:
Leave an empty line between the `var`s declaration block and the rest of the code
```js
var deferred = $q.defer();

if () {

}
```